pramen/core/src/main/scala/za/co/absa/pramen/core/process/ProcessRunnerImpl.scala

```diff
@@ -88,19 +106,43 @@ class ProcessRunnerImpl(includeOutputLines: Int,
   }
 
   private[core] def processReader(reader: BufferedReader,
-                            buffer: Option[CircularBuffer[String]],
-                            prefix: String,
-                            logEnabled: Boolean): Unit = {
+                                  buffer: Option[CircularBuffer[String]],
+                                  prefix: String,
+                                  logEnabled: Boolean): Unit = {
     val cmdOutputLogger: Logger = LoggerFactory.getLogger(prefix)
+    val filterRs = outputFilterRegEx.map(_.r)
+    val recCountR = recordCountRegEx.map(_.r)
+    val zeroRecordsSuccessR = zeroRecordsSuccessRegEx.map(_.r)
+    val failureR = failureRegEx.map(_.r)
 
     try {
       var line: String = reader.readLine
```

Comment:
I'm thinking if some pure functions could be extracted from this block (e.g. `extractRecordCount`, `extractFailure`, `extractZeroRecordSuccess`) that would operate on the line String but variable assignment would get handled inside the block. But I understand the limitations - we have a lot of shared variables (the regexes) and we need to modify the `var`s like recordCount and failureFound etc. Could be hard to extract some meaningful functions.

Reply:
Good point, the method right now looks quite ugly - it has many concerns and looks a bit 'scriptish'. I'll try to improve it.